
        <!DOCTYPE html>
        <html lang="en-us">
        <head>
            <meta charset="UTF-8">
            <title>AR_0859</title>
            <style>
            h1 {
            font-family: Arial, Helvetica, sans-serif;
            font-size: 1.5em;
            text-align: center;
            color: #1e5c85
            }

            h2 {
            font-family: Arial, Helvetica, sans-serif;
            font-size: 1em;
            margin-top: 1em;
            margin-bottom: 0.4em;
            text-align: left;
            color: #D6672E 
            }

            table {
            border-collapse: collapse;
            border: none;                                       
            }
            
            th, td {
            text-align: left;
            padding-left: 5px;
            padding-right: 5px;
            padding-top: 1px;
            padding-bottom: 1px;
            border: none;
            }
            
            table.dataframe tr:nth-child(even) {background-color: #f2f2f2;}
            
            table.dataframe th {
            background-color: #1e5c85;
            color: white;
            }
            
            table.custom tr:nth-child(odd) {background-color: #f2f2f2;}
            
            table.custom th {
            background-color: #1e5c85;
            color: white;
            }
            </style>
        </head>
        <body>
        <header>
        <h1>QC Summary</h1>
        </header>
        <hr>
        <article>
        <h2>Sample Information</h2>
        <table class="custom">
        <tr>
            <td>Report date:</td>
            <td>2023-05-19</td>
        </tr>
        <tr>
            <td>Laboratory ID:</td>
            <td>AR_0859</td>
        </tr>
        <tr>
            <td>Pipeline:</td>
            <td>C-BIRD v1.0.0</td>
        </tr>
        </table>
        <h2>Sequencing Statistics</h2>

        <table class="custom">
        <tr>
        <td>Sequencing depth:</td>
        <td>73.54</td>
        </tr>
        <tr>
        <td>PhiX ratio:</td>
        <td>0.28934%</td>
        </tr>
        <tr>
        <td>Estimated genome ratio:</td>
        <td>1.02</td>
        </tr>
        </table>
        <h2>Quality Trimming & Filtering</h2>
        <table border="1" class="dataframe">
  <thead>
    <tr style="text-align: left;">
      <th>General</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>sequencing:</td>
      <td>paired end (251 cycles + 251 cycles)</td>
    </tr>
    <tr>
      <td>mean length before filtering:</td>
      <td>221bp, 222bp</td>
    </tr>
    <tr>
      <td>mean length after filtering:</td>
      <td>207bp, 190bp</td>
    </tr>
    <tr>
      <td>duplication rate:</td>
      <td>0.040504%</td>
    </tr>
    <tr>
      <td>Insert size peak:</td>
      <td>251</td>
    </tr>
  </tbody>
</table>
        <p></p>
        <table border="1" class="dataframe">
  <thead>
    <tr style="text-align: left;">
      <th></th>
      <th>Before</th>
      <th>After</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>total reads:</td>
      <td>1.708486 M</td>
      <td>1.455704 M</td>
    </tr>
    <tr>
      <td>total bases:</td>
      <td>378.705623 M</td>
      <td>289.926074 M</td>
    </tr>
    <tr>
      <td>Q20 bases:</td>
      <td>343.923060 M (90.815409%)</td>
      <td>283.954871 M (97.940439%)</td>
    </tr>
    <tr>
      <td>Q30 bases:</td>
      <td>327.612033 M (86.508362%)</td>
      <td>280.166970 M (96.633934%)</td>
    </tr>
    <tr>
      <td>GC content:</td>
      <td>51.129830%</td>
      <td>50.507979%</td>
    </tr>
  </tbody>
</table>
        <p></p>
        <table border="1" class="dataframe">
  <thead>
    <tr style="text-align: left;">
      <th>Quality Filtering</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>reads passed filters:</td>
      <td>1.455704 M (85.204327%)</td>
    </tr>
    <tr>
      <td>reads with low quality:</td>
      <td>117.598000 K (6.883170%)</td>
    </tr>
    <tr>
      <td>reads with too many N:</td>
      <td>0 (0.000000%)</td>
    </tr>
    <tr>
      <td>reads too short:</td>
      <td>135.184000 K (7.912503%)</td>
    </tr>
  </tbody>
</table>
        <p></p>
        <table border="1" class="dataframe">
  <thead>
    <tr style="text-align: left;">
      <th>Adapter or bad ligation of read1</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Sequence</td>
      <td>Occurrences</td>
    </tr>
    <tr>
      <td>C</td>
      <td>2</td>
    </tr>
    <tr>
      <td>G</td>
      <td>18</td>
    </tr>
    <tr>
      <td>AGT</td>
      <td>3</td>
    </tr>
    <tr>
      <td>GGC</td>
      <td>2</td>
    </tr>
    <tr>
      <td>AGATCGGG</td>
      <td>2</td>
    </tr>
    <tr>
      <td>other adapter sequences</td>
      <td>119</td>
    </tr>
  </tbody>
</table>
        <p></p>
        <table border="1" class="dataframe">
  <thead>
    <tr style="text-align: left;">
      <th>Adapter or bad ligation of read2</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Sequence</td>
      <td>Occurrences</td>
    </tr>
    <tr>
      <td>A</td>
      <td>77</td>
    </tr>
    <tr>
      <td>G</td>
      <td>335</td>
    </tr>
    <tr>
      <td>T</td>
      <td>70</td>
    </tr>
    <tr>
      <td>AG</td>
      <td>197</td>
    </tr>
    <tr>
      <td>AGG</td>
      <td>472</td>
    </tr>
    <tr>
      <td>AGT</td>
      <td>71</td>
    </tr>
    <tr>
      <td>AGATG</td>
      <td>196</td>
    </tr>
    <tr>
      <td>AGATCGG</td>
      <td>524</td>
    </tr>
    <tr>
      <td>AGATCGGA</td>
      <td>81</td>
    </tr>
    <tr>
      <td>AGATCGGG</td>
      <td>654</td>
    </tr>
    <tr>
      <td>AGATCGGAAG</td>
      <td>64</td>
    </tr>
    <tr>
      <td>other adapter sequences</td>
      <td>3460</td>
    </tr>
  </tbody>
</table>
        <p></p>
        <h2>Taxonomy Report</h2>
        <table border="1" class="dataframe">
  <thead>
    <tr style="text-align: left;">
      <th>name</th>
      <th>taxonomy_id</th>
      <th>taxonomy_lvl</th>
      <th>kraken_assigned_reads</th>
      <th>added_reads</th>
      <th>new_est_reads</th>
      <th>fraction_total_reads</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Escherichia coli</td>
      <td>562</td>
      <td>S</td>
      <td>277927</td>
      <td>423017</td>
      <td>700944</td>
      <td>0.98764</td>
    </tr>
  </tbody>
</table>
        <h2>Assembly Statistics</h2>
        <table border="1" class="dataframe">
  <thead>
    <tr style="text-align: left;">
      <th>Assembly</th>
      <th>AR_0859_scaffolds_trim</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td># contigs (&gt;= 0 bp)</td>
      <td>84.00</td>
    </tr>
    <tr>
      <td># contigs (&gt;= 1000 bp)</td>
      <td>69.00</td>
    </tr>
    <tr>
      <td># contigs (&gt;= 5000 bp)</td>
      <td>45.00</td>
    </tr>
    <tr>
      <td># contigs (&gt;= 10000 bp)</td>
      <td>42.00</td>
    </tr>
    <tr>
      <td># contigs (&gt;= 25000 bp)</td>
      <td>37.00</td>
    </tr>
    <tr>
      <td># contigs (&gt;= 50000 bp)</td>
      <td>25.00</td>
    </tr>
    <tr>
      <td>Total length (&gt;= 0 bp)</td>
      <td>5252010.00</td>
    </tr>
    <tr>
      <td>Total length (&gt;= 1000 bp)</td>
      <td>5241699.00</td>
    </tr>
    <tr>
      <td>Total length (&gt;= 5000 bp)</td>
      <td>5182561.00</td>
    </tr>
    <tr>
      <td>Total length (&gt;= 10000 bp)</td>
      <td>5161651.00</td>
    </tr>
    <tr>
      <td>Total length (&gt;= 25000 bp)</td>
      <td>5088718.00</td>
    </tr>
    <tr>
      <td>Total length (&gt;= 50000 bp)</td>
      <td>4684764.00</td>
    </tr>
    <tr>
      <td># contigs</td>
      <td>84.00</td>
    </tr>
    <tr>
      <td>Largest contig</td>
      <td>686739.00</td>
    </tr>
    <tr>
      <td>Total length</td>
      <td>5252010.00</td>
    </tr>
    <tr>
      <td>GC (%)</td>
      <td>50.37</td>
    </tr>
    <tr>
      <td>N50</td>
      <td>189030.00</td>
    </tr>
    <tr>
      <td>N75</td>
      <td>108686.00</td>
    </tr>
    <tr>
      <td>L50</td>
      <td>8.00</td>
    </tr>
    <tr>
      <td>L75</td>
      <td>17.00</td>
    </tr>
    <tr>
      <td># N's per 100 kbp</td>
      <td>3.77</td>
    </tr>
  </tbody>
</table>
        <h2>Assembly Assesment</h2>
        <table border="1" class="dataframe">
  <thead>
    <tr style="text-align: left;">
      <th>description</th>
      <th>results</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>domain</td>
      <td>prokaryota</td>
    </tr>
    <tr>
      <td>one_line_summary</td>
      <td>C:100.0%[S:99.2%,D:0.8%],F:0.0%,M:0.0%,n:124</td>
    </tr>
    <tr>
      <td>Complete</td>
      <td>100.0</td>
    </tr>
    <tr>
      <td>Single copy</td>
      <td>99.2</td>
    </tr>
    <tr>
      <td>Multi copy</td>
      <td>0.8</td>
    </tr>
    <tr>
      <td>Fragmented</td>
      <td>0.0</td>
    </tr>
    <tr>
      <td>Missing</td>
      <td>0.0</td>
    </tr>
    <tr>
      <td>n_markers</td>
      <td>124</td>
    </tr>
    <tr>
      <td>Number of scaffolds</td>
      <td>84</td>
    </tr>
    <tr>
      <td>Number of contigs</td>
      <td>86</td>
    </tr>
    <tr>
      <td>Total length</td>
      <td>5252010</td>
    </tr>
    <tr>
      <td>Percent gaps</td>
      <td>0.004%</td>
    </tr>
    <tr>
      <td>Scaffold N50</td>
      <td>189030</td>
    </tr>
    <tr>
      <td>Contigs N50</td>
      <td>189030</td>
    </tr>
  </tbody>
</table>
        <p></p>
        </article>
        <hr>
        <footer>
        <p><i>This QC summary is created by <a href="https://github.com/Kincekara/C-BIRD">C-BIRD v1.0.0</a> bioinformatics pipeline.</i></p>
        </footer>
    </body>
    </html>
    