<!DOCTYPE html>
<html lang="en-us">
<head>
    <meta charset="UTF-8">
    <title>AR-859</title>
    <style>
    h1 {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.5em;
    text-align: center;
    color: #1e5c85
    }

    h2 {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1em;
    margin-top: 1em;
    margin-bottom: 0.4em;
    text-align: left;
    color: #D6672E 
    }

    table {
    border-collapse: collapse;
    border: none;                                       
    }
    
    th, td {
    text-align: left;
    padding-left: 5px;
    padding-right: 5px;
    padding-top: 1px;
    padding-bottom: 1px;
    border: none;
    }
    
    table.dataframe tr:nth-child(even) {background-color: #f2f2f2;}
    
    table.dataframe th {
    background-color: #1e5c85;
    color: white;
    }
    
    table.custom tr:nth-child(odd) {background-color: #f2f2f2;}
    
    table.custom th {
    background-color: #1e5c85;
    color: white;
    }
    </style>
</head>
<body>
<header>
<h1>QC Summary</h1>
</header>
<hr>
<article>
<h2>Sample Information</h2>
<table class="custom">
<tr>
    <td>Report date:</td>
    <td>2022-12-19</td>
</tr>
<tr>
    <td>Laboratory ID:</td>
    <td>AR-859</td>
</tr>
<tr>
    <td>Pipeline:</td>
    <td>C-BIRD v1.0.0</td>
</tr>
</table>
<h2>Sequencing Statistics</h2>

<table class="custom">
<tr>
<td>Sequencing depth:</td>
<td>67.26</td>
</tr>
<tr>
<td>PhiX ratio:</td>
<td>0.00000%</td>
</tr>
<tr>
<td>Estimated genome ratio:</td>
<td>1.02</td>
</tr>
</table>
<h2>Quality Trimming & Filtering</h2>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: left;">
<th>General</th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>sequencing:</td>
<td>paired end (151 cycles + 151 cycles)</td>
</tr>
<tr>
<td>mean length before filtering:</td>
<td>148bp, 148bp</td>
</tr>
<tr>
<td>mean length after filtering:</td>
<td>146bp, 145bp</td>
</tr>
<tr>
<td>duplication rate:</td>
<td>0.636904%</td>
</tr>
<tr>
<td>Insert size peak:</td>
<td>151</td>
</tr>
</tbody>
</table>
<p></p>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: left;">
<th></th>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>total reads:</td>
<td>2.328764 M</td>
<td>2.110068 M</td>
</tr>
<tr>
<td>total bases:</td>
<td>346.376945 M</td>
<td>307.832627 M</td>
</tr>
<tr>
<td>Q20 bases:</td>
<td>332.791294 M (96.077784%)</td>
<td>304.012061 M (98.758882%)</td>
</tr>
<tr>
<td>Q30 bases:</td>
<td>328.307719 M (94.783364%)</td>
<td>302.333443 M (98.213580%)</td>
</tr>
<tr>
<td>GC content:</td>
<td>50.775723%</td>
<td>50.641747%</td>
</tr>
</tbody>
</table>
<p></p>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: left;">
<th>Quality Filtering</th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>reads passed filters:</td>
<td>2.110068 M (90.608924%)</td>
</tr>
<tr>
<td>reads with low quality:</td>
<td>78.412000 K (3.367108%)</td>
</tr>
<tr>
<td>reads with too many N:</td>
<td>0 (0.000000%)</td>
</tr>
<tr>
<td>reads too short:</td>
<td>140.284000 K (6.023968%)</td>
</tr>
</tbody>
</table>
<p></p>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: left;">
<th>Adapter or bad ligation of read1</th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>Sequence</td>
<td>Occurrences</td>
</tr>
<tr>
<td>C</td>
<td>4</td>
</tr>
<tr>
<td>other adapter sequences</td>
<td>250</td>
</tr>
</tbody>
</table>
<p></p>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: left;">
<th>Adapter or bad ligation of read2</th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>Sequence</td>
<td>Occurrences</td>
</tr>
<tr>
<td>C</td>
<td>27</td>
</tr>
<tr>
<td>other adapter sequences</td>
<td>1389</td>
</tr>
</tbody>
</table>
<p></p>
<h2>Taxonomy Report</h2>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: left;">
<th>name</th>
<th>taxonomy_id</th>
<th>taxonomy_lvl</th>
<th>kraken_assigned_reads</th>
<th>added_reads</th>
<th>new_est_reads</th>
<th>fraction_total_reads</th>
</tr>
</thead>
<tbody>
<tr>
<td>Escherichia coli</td>
<td>562</td>
<td>S</td>
<td>418416</td>
<td>606289</td>
<td>1024705</td>
<td>0.98979</td>
</tr>
</tbody>
</table>
<h2>Assembly Statistics</h2>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: left;">
<th>Assembly</th>
<th>AR_859_scaffolds_trim</th>
</tr>
</thead>
<tbody>
<tr>
<td># contigs (&gt;= 0 bp)</td>
<td>79.00</td>
</tr>
<tr>
<td># contigs (&gt;= 1000 bp)</td>
<td>68.00</td>
</tr>
<tr>
<td># contigs (&gt;= 5000 bp)</td>
<td>47.00</td>
</tr>
<tr>
<td># contigs (&gt;= 10000 bp)</td>
<td>42.00</td>
</tr>
<tr>
<td># contigs (&gt;= 25000 bp)</td>
<td>36.00</td>
</tr>
<tr>
<td># contigs (&gt;= 50000 bp)</td>
<td>28.00</td>
</tr>
<tr>
<td>Total length (&gt;= 0 bp)</td>
<td>5230393.00</td>
</tr>
<tr>
<td>Total length (&gt;= 1000 bp)</td>
<td>5222632.00</td>
</tr>
<tr>
<td>Total length (&gt;= 5000 bp)</td>
<td>5167554.00</td>
</tr>
<tr>
<td>Total length (&gt;= 10000 bp)</td>
<td>5127976.00</td>
</tr>
<tr>
<td>Total length (&gt;= 25000 bp)</td>
<td>5019768.00</td>
</tr>
<tr>
<td>Total length (&gt;= 50000 bp)</td>
<td>4728766.00</td>
</tr>
<tr>
<td># contigs</td>
<td>79.00</td>
</tr>
<tr>
<td>Largest contig</td>
<td>481060.00</td>
</tr>
<tr>
<td>Total length</td>
<td>5230393.00</td>
</tr>
<tr>
<td>GC (%)</td>
<td>50.36</td>
</tr>
<tr>
<td>N50</td>
<td>192348.00</td>
</tr>
<tr>
<td>N75</td>
<td>115270.00</td>
</tr>
<tr>
<td>L50</td>
<td>10.00</td>
</tr>
<tr>
<td>L75</td>
<td>18.00</td>
</tr>
<tr>
<td># N's per 100 kbp</td>
<td>20.59</td>
</tr>
</tbody>
</table>
<h2>Assembly Assesment</h2>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: left;">
<th>description</th>
<th>results</th>
</tr>
</thead>
<tbody>
<tr>
<td>domain</td>
<td>prokaryota</td>
</tr>
<tr>
<td>one_line_summary</td>
<td>C:100.0%[S:99.2%,D:0.8%],F:0.0%,M:0.0%,n:124</td>
</tr>
<tr>
<td>Complete</td>
<td>100.0</td>
</tr>
<tr>
<td>Single copy</td>
<td>99.2</td>
</tr>
<tr>
<td>Multi copy</td>
<td>0.8</td>
</tr>
<tr>
<td>Fragmented</td>
<td>0.0</td>
</tr>
<tr>
<td>Missing</td>
<td>0.0</td>
</tr>
<tr>
<td>n_markers</td>
<td>124</td>
</tr>
<tr>
<td>Number of scaffolds</td>
<td>79</td>
</tr>
<tr>
<td>Number of contigs</td>
<td>90</td>
</tr>
<tr>
<td>Total length</td>
<td>5230393</td>
</tr>
<tr>
<td>Percent gaps</td>
<td>0.021%</td>
</tr>
<tr>
<td>Scaffold N50</td>
<td>192348</td>
</tr>
<tr>
<td>Contigs N50</td>
<td>169456</td>
</tr>
</tbody>
</table>
<p></p>
</article>
<hr>
<footer>
<p><i>This QC summary is created by <a href="https://github.com/Kincekara/C-BIRD">C-BIRD v1.0.0</a> bioinformatics pipeline.</i></p>
</footer>
</body>
</html>
